modules/internal/DefaultRectangular.chpl

```diff
@@ -809,6 +1013,241 @@ module DefaultRectangular {
         yield dsiAccess(i);
     }
 
+    //
+    // Potentially multi-ddata iterators (locale models with sublocales)
+    //
+    iter these(tasksPerLocale:int = dataParTasksPerLocale,
+               ignoreRunning:bool = dataParIgnoreRunningTasks,
+               minIndicesPerTask:int = dataParMinGranularity)
+      ref where !defRectSimpleDData {
+      if rank == 1 {
+        if !dom.stridable {
+          const first = getDataIndex(dom.dsiLow, getChunked=false);
+          const second = getDataIndex(dom.dsiLow+1, getChunked=false);
+          const step = (second-first);
+          if mdNumChunks == 1 {
+            const last = first + (dom.dsiNumIndices-1) * step;
+            const dd = theDataChunk(0);
+            for i in chpl_direct_pos_stride_range_iter(first, last, step) {
+              yield dd(i);
+            }
+          } else if dom.dsiNumIndices > 0 {
+            const (chunkLo, _) = getDataIndex(dom.dsiLow);
+            const (chunkHi, _) = getDataIndex(dom.dsiHigh);
+
+            // first chunk
+            {
+              const dd = theDataChunk(chunkLo);
+              const (_, hi) = mdChunk2Ind(chunkLo);
```

Comment:
I take it that mdChunk2Ind doesn't return dom.dsiLow/dsiHigh correctly? Is that something that could be done at a later time, or is it technically not feasible?